File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cs
public class Solution {
    public IList<int> FindDuplicates(int[] nums) {
        bool[] dupe_list = new bool[nums.Length + 1];
        IList<int> return_list = new List<int>();
        foreach(int e in nums){
            if(dupe_list[e])
                return_list.Add(e);
            else
                dupe_list[e] = true;
        }
        
        return return_list;
    }
}

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.py
class Node:
    data: int = -1
    left = None
    right = None
    
    def __init__(self, data: int):
        self.data = data

    def Add(self, n) -> bool:
        if n.data > self.data:
            if self.right is None:
                self.right = n
                return True
            else:
                return self.right.Add(n)
        if n.data < self.data:
            if self.left is None:
                self.left = n
                return True
            else:
                return self.left.Add(n)
        if n.data == self.data:
            return False
        

class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        return_list = []
        tree = Node(nums.pop())
        
        while len(nums) > 0:
            test_dupe = nums.pop()
            if not tree.Add(Node(test_dupe)):
                return_list.append(test_dupe)
        
        return return_list
    